lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb

```diff
@@ -5,55 +5,73 @@ module Auth
     module FirebaseAppDistributionAuthClient
       TOKEN_CREDENTIAL_URI = "https://oauth2.googleapis.com/token"
 
-      def fetch_auth_token(google_service_path)
+      # Returns the auth token for any of the auth methods (Firebase CLI token,
+      # Google service account, firebase-tools). To ensure that a specific
+      # auth method is used, unset all other auth variables/parameters to nil/empty
+      #
+      # args
+      #   google_service_path - Absolute path to the Google service account file
+      #   firebase_cli_token - Firebase CLI refresh token from login action or
+      #                        CI environment
+      #
+      # env variables
+      #   GOOGLE_APPLICATION_CREDENTIALS - see google_service_path
+      #   FIREBASE_TOKEN - see firebase_cli_token
+      #
+      # Crashes if given invalid or missing credentials
+      def fetch_auth_token(google_service_path, firebase_cli_token)
         if !google_service_path.nil? && !google_service_path.empty?
-          service_account(google_service_path)
-        elsif ENV["FIREBASE_TOKEN"]
-          firebase_token(ENV["FIREBASE_TOKEN"])
-        elsif ENV["GOOGLE_APPLICATION_CREDENTIALS"]
-          service_account(ENV["GOOGLE_APPLICATION_CREDENTIALS"])
+          UI.message("Authenticating with --service_credentials_file path parameter")
+          token = service_account(google_service_path)
+        elsif !firebase_cli_token.nil? && !firebase_cli_token.empty?
+          UI.message("Authenticating with --firebase_cli_token parameter")
+          token = firebase_token(firebase_cli_token)
+        elsif !ENV["FIREBASE_TOKEN"].nil? && !ENV["FIREBASE_TOKEN"].empty?
+          UI.message("Authenticating with FIREBASE_TOKEN environment variable")
+          token = firebase_token(ENV["FIREBASE_TOKEN"])
+        elsif !ENV["GOOGLE_APPLICATION_CREDENTIALS"].nil? && !ENV["GOOGLE_APPLICATION_CREDENTIALS"].empty?
+          UI.message("Authenticating with GOOGLE_APPLICATION_CREDENTIALS environment variable")
+          token = service_account(ENV["GOOGLE_APPLICATION_CREDENTIALS"])
         elsif (refresh_token = refresh_token_from_firebase_tools)
-          firebase_token(refresh_token)
+          UI.message("Authenticating with the firebase login:ci command")
```

Comment:
I think we should be explicit that this is a fallback method because no other authentication method was used. So maybe something like:

"No authentication method specified. Using cached Firebase CLI credentials."

@lfkellogg thoughts?

Reply:
Ooo, yeah I like that!